### ISSUE TITLE
enable service-serving-cert-signer by default

### ISSUE THREAD

**Issue by deads2k** [openshift/openshift-ansible#2345]
Commands like `oadm router` want to take advantage of the service-serving-cert-signing controller.  This requires that 
```yaml
controllerConfig:
  serviceServingCert:
    signer:
      certFile: service-signer.crt
      keyFile: service-signer.key
```

in the master-config.yaml be configured with a signer.  This signs for internal DNS names and can be created using a command like:

```bash
oadm ca create-signer-cert --cert=service-signer.crt --key=service-signer.key --serial=service-signer-serial.txt
```

https://github.com/openshift/openshift-docs/pull/2324 has docs about using it.

**Comment by knobunc** [comment]
@pecameron you need this fix.

**Comment by pecameron** [comment]
@knobunc OK

**Comment by eparis** [comment]
changing to p0. unable to deploy 3.3 routers without.

**Comment by itewk** [comment]
For anyone wondering who stumbles on this when you are debuging your OCP 3.3 cluster after an upgrade to 3.2 and wondering why it keeps choking on service-signer.crt missing. Be sure you are using the 3.3.30-1 or later tag of the ansible scripts, anything before that doesn't install the service-signer.crt which is required for a successful upgrade.

**Comment by itewk** [comment]
looks like 3.3.30-1 doesn't work either, trying master.

on 3.3.30-1 the "Upgrade master packages" play happens just before the "Determine if service signer cert must be created" play and that "Upgrade master packages" play fails on the "Restart node service" task because the cert does not exist.

**Comment by itewk** [comment]
master doesn't work either

**Comment by itewk** [comment]
patched: https://github.com/openshift/openshift-ansible/pull/2557

**Comment by dagwieers** [comment]
This bit us when doing the OSE v3.2 tot OCP v3.3 upgrade :-/ The Ansible-driven advanced upgrade process fails to get to the finish successfully. Probably warrants an official fix to Red Hat OCP v3.3 ?

**Comment by detiber** [comment]
@itewk @dagwieers what versions of Ansible are you using? I'm wondering if there is version dependent behavior causing the handlers to be triggered at different times depending on the  version in use.

**Comment by itewk** [comment]
@detiber 

Steps to Reproduce:
The first attempt at upgrade was using the playbooks shipped with the 3.3 RPMs. That version is 3.3.28-1. That version does not contain the Ansible tasks to install the certs if needed. It apeared from that upgrade run that the upgrade worked. But then on a run of the playbooks/byo/config.yml the services attempted to restart and ran into the error.

We then tried using the latest ansible playbooks for 3.3, 3.3.30-1, those scripts have the tasks to install the certs if needed, but not in the correct order. To resolve the issue we had to update the scripts to install the certs before doing the master package upgrade. Patch link below.

playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade.yml (3.3.28-1, shipped with 3.3 RPMs)
playbooks/byo/update.yml (3.3.28-1, shipped with 3.3 RPMs)
run failed due to missing certs
download 3.3.30-1 scripts from github
playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade.yml (3.3.30-1)
Failed for same error because step to install certs is just after the package upgrade which is the step that fails due to restart
apply patch (#2557)
playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade.yml (3.3.30-1 with patch)
success
playbooks/byo/update.yml (3.3.30-1 with patch)
success
